src/StockServiceManager.php

```diff
@@ -29,13 +30,30 @@ class StockServiceManager implements StockServiceManagerInterface, StockTransact
   protected $configFactory;
 
   /**
+   * The current store.
+   *
+   * @var \Drupal\commerce_store\Entity\Store
+   */
+  protected $currentStore;
+
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $currentUser;
+
+  /**
    * Constructs a StockServiceManager object.
    *
    * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
    *   The config factory.
    */
   public function __construct(ConfigFactoryInterface $config_factory) {
     $this->configFactory = $config_factory;
```

Comment:
I think the issue is that the product edit form may not have a context, if so this will be an issue with where ever ever we do this.